### PR TITLE
pldm: Support password prompt for graphical SMS menu

### PR DESCRIPTION
**Pull request by jaypadath**
This supports the following password prompts for PFW (Platform Firmware) provided graphical SMS (System Management Services (SMS) menu

In system configurations that include a graphics adapter assigned to a partition, PFW can display the SMS menus on the console allowing users to interact with PFW using a keybaord. This menu requires as they can be used to change partition configurations (e.g. Boot option, I/O device info etc.)

Supported password prompt functions:
1. The ability to prompt for a username/password and provide authentication
2. The ability to change a password when the current password is expired

Change-Id: Ie42c5306e6447f65064bb8cb9480a3cde0e4ed4e

### CODE REVIEW

**Review by manojkiraneda — APPROVED** [review]
The changes are pretty much the same as 1060, and this is just a rebase. This code would definitely need a lot of refactor.

File: oem/ibm/libpldmresponder/file_io.cpp [hunk 1]
```diff
@@ -939,15 +940,38 @@ Response Handler::writeFileByTypeFromMemory(const pldm_msg* request,
...
+    try
+    {
+        handler = getHandlerByType(fileType, fileHandle, instanceIdDb,
+                                   Handler::handler);
+    }
+    catch (const InternalFailure& e)
+    {
+        error("post write call back failed with file type : {FILE_TYPE} and"
+              " error : {ERR_EXCEP}",
+              "FILE_TYPE", fileType, "ERR_EXCEP", e.what());
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/file_io.cpp:970]: lg2 new format "e"

File: oem/ibm/libpldmresponder/file_io.cpp [hunk 2]
```diff
@@ -939,15 +940,38 @@ Response Handler::writeFileByTypeFromMemory(const pldm_msg* request,
...
                                            size_t payloadLength)
 {
     return rwFileByTypeIntoMemory(PLDM_READ_FILE_BY_TYPE_INTO_MEMORY, request,
                                   payloadLength, oemPlatformHandler,
-                                  sharedAIORespDataobj);
+                                  sharedAIORespDataobj, instanceIdDb);
+}
+
+void Handler::postWriteCallBack(
+    const uint16_t fileType, const uint32_t fileHandle,
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/file_io.cpp:955]: can we pass these params as a reference?

File: oem/ibm/libpldmresponder/file_io_type_smsmenu.hpp [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+    virtual int postDataTransferCallBack(bool /*IsWriteToMemOp*/,
+                                         uint32_t /*length*/)
+    {
+        return PLDM_ERROR_UNSUPPORTED_PLDM_CMD;
+    }
+
+    virtual void
+        postWriteAction(const uint16_t fileType, const uint32_t fileHandle,
+                        const struct fileack_status_metadata& metaDataObj);
```
Comment (Pavithrab7) [inline oem/ibm/libpldmresponder/file_io_type_smsmenu.hpp:99]: can we have doxygen for this?

**Review by Pavithrab7 — CHANGES_REQUESTED** [review]
Jay, did we use clang-form-18 on these files?

File: oem/ibm/libpldmresponder/file_io.cpp (hunk 1 above)
Comment (jaypadath) [reply oem/ibm/libpldmresponder/file_io.cpp:970]: Done

File: oem/ibm/libpldmresponder/file_io.cpp (hunk 2 above)
Comment (jaypadath) [reply oem/ibm/libpldmresponder/file_io.cpp:955]: Done

File: oem/ibm/libpldmresponder/file_io_type_smsmenu.hpp (hunk 3 above)
Comment (jaypadath) [reply oem/ibm/libpldmresponder/file_io_type_smsmenu.hpp:99]: We already have doxygen for all of these virtual functions in file_io_by_type.hpp

**Comment by jaypadath** [comment]
Ran clang-form-18 on all these files

**Review by Pavithrab7 — APPROVED** [review]

**Pull request merged**